docker.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+# plugin to set "dnf" proxy settings for ProxyMan
+# privileges has to be set by the process which starts this script
+
+CONF_FILE="/etc/systemd/system/docker.service.d/http-proxy.conf"
+
+
+list_proxy() {
+    # inefficient way as the file is read twice.. think of some better way
+    echo -e "${bold}docker proxy settings: ${normal}"
+    if [ ! -e "$CONF_FILE" ]; then
+        echo -e "${red}None${normal}"
+        return
+    else
+        lines="$(cat $CONF_FILE | grep proxy -i | wc -l)"
+        if [ "$lines" -gt 0 ]; then
+            cat $CONF_FILE | grep proxy -i | sed -e "s/Environment=//g" -e "s/\_/\ /g"
+        else
+            echo -e "${red}None${normal}"
+        fi
+    fi
+}
+
+unset_proxy() {
+    if [ ! -e "$CONF_FILE" ]; then
+        return
+    fi
+    if [ "$(cat $CONF_FILE | grep proxy -i | wc -l)" -gt 0 ]; then
+        sudo rm $CONF_FILE
+        sudo service docker restart
+    fi
+}
+
+set_proxy() {
+    unset_proxy
+    if [ ! -e "$CONF_FILE" ]; then
+        sudo touch "$CONF_FILE"
+    fi
+
+    local stmt=""
+    if [ "$use_auth" = "y" ]; then
+        stmt="${username}:${password}@"
+    fi
+
+    echo -n "" > docker_config.tmp
+    echo "[Service]" >> docker_config.tmp
+    echo "Environment=\"HTTP_PROXY=http://${stmt}${http_host}:${http_port}\";" \
+         >> docker_config.tmp
+    echo "Environment=\"HTTP_PROXY=http://${stmt}${http_host}:${http_port}\";" \
```

Comment:
Testing initially, it didn't work for me.
I had to do two changes in the conf file to make it work.
* add a `/` at the end of URL
* Add multiple proxy schemes in one line as mentioned [here](https://docs.docker.com/config/daemon/systemd/#httphttps-proxy)
  This one seems to re-initialize a variable.